linera-rpc/src/grpc_network.rs

```diff
@@ -516,7 +529,10 @@ where
         let request = request.into_inner().try_into()?;
         debug!(?request, "Handling cross-chain request");
         match self.state.clone().handle_cross_chain_request(request).await {
-            Ok(actions) => self.handle_network_actions(actions),
+            Ok(actions) => {
+                SERVER_REQUEST_SUCCESS.with_label_values(&[]).inc();
```

Comment:
Right, these ones aren't even coming from clients.
Why not measure latency and count in here, too, and add the request type as a label?

Reply:
My thinking was that we need to measure the total latency no matter what's happening in the server. These different types of requests are called by something. If there's a latency issue in that part of the code for some reason, we would miss it. So doing it like this could leave us open to some blind spots. Doing it with the `tower` middleware makes us be sure we're capturing everything. Does that make sense or do you think I'm missing something?

Maybe we could have a different metric for it? Like a `server_request_latency_per_request_type` with labels for each one, or one metric per type? I personally prefer the former. Let me know what you think!

Reply:
Sure, maybe we should have both.
(I'd also use labels for the different request types, yes.)